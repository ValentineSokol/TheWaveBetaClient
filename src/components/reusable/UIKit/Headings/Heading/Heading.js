import React from "react";
import './Heading.scss';

const Heading = ({ children, size }) => {
    const HeadingTag = `h${size}`;
    return <HeadingTag className='Heading'>{children}</HeadingTag>

};
Heading.defaultProps = {
    size: '1'
}

export default Heading;