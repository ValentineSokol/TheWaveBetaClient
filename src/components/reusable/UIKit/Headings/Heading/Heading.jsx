import React from 'react';
import './Heading.scss';

function Heading({ children, size }) {
  const HeadingTag = `h${size}`;
  return <HeadingTag className="Heading">{children}</HeadingTag>;
}

export default Heading;
