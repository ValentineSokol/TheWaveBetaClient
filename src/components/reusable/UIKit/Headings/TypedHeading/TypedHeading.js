import React from 'react';
import Typed from "../../../Typed";
import Heading from "../Heading/Heading";

const TypedHeading = ({ headingStrings, size, loop }) => (
    <Heading size={size}>
        {
            headingStrings?
            <Typed strings={headingStrings} loop={loop}/>
            :
            null

        }
    </Heading> );
TypedHeading.defaultProps = {
    size: '1',
    loop: false
}
export default TypedHeading;