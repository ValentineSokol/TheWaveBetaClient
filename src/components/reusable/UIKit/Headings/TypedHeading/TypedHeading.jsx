import React from 'react';
import PropTypes from 'prop-types';
import Typed from '../../../Typed';
import Heading from '../Heading/Heading';

function TypedHeading({ headingStrings, size, loop }) {
  return (
    <Heading size={size}>
      {
            headingStrings
              ? <Typed strings={headingStrings} loop={loop} />
              : null

        }
    </Heading>
  );
}

TypedHeading.propTypes = {
  headingStrings: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.arrayOf(PropTypes.string),
  ]).isRequired,
  size: PropTypes.oneOf(['1', '2', '3', '4', '5', '6']),
  loop: PropTypes.bool,
};
TypedHeading.defaultProps = {
  size: '1',
  loop: false,
};
export default TypedHeading;
