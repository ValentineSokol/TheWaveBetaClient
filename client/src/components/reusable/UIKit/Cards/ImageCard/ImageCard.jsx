import React from 'react';
import Card from '../Card/Card';
import TypedHeading from '../../Headings/TypedHeading/TypedHeading';
import Heading from '../../Headings/Heading/Heading';
import './ImageCard.scss';

function ImageCard({
  image, imageAlt, headingStrings, headingSize, text, width,
}) {
  return (
    <Card width={width}>
      <div className="ContentWrapper">
        <img data-testid="image" src={image} alt={imageAlt} />
        <div>
          {
                 Array.isArray(headingStrings)
                   ? <TypedHeading headingStrings={headingStrings} size={headingSize} />
                   : <Heading size={headingSize}>{headingStrings}</Heading>
             }
          <p data-testid="text-content">{text}</p>
        </div>
      </div>
    </Card>
  );
}
export default ImageCard;
