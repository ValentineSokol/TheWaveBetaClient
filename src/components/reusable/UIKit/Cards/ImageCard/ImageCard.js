import React from 'react';
import Card from "../Card/Card";
import TypedHeading from "../../Headings/TypedHeading/TypedHeading";
import Heading from "../../Headings/Heading/Heading";
import './ImageCard.scss';

const ImageCard = ({ image, round, imageAlt, headingStrings, headingSize, text, width, padding }) => (
    <Card width={width} padding={padding}>
        <div className='ContentWrapper'>
         <img className={round && 'RoundImage'} data-testid='image' src={image} alt={imageAlt} />
         <div>
             {
                 Array.isArray(headingStrings)?
                     <TypedHeading headingStrings={headingStrings} size={headingSize} />
                     :
                     <Heading size={headingSize}>{headingStrings}</Heading>
             }
            <p data-testid='text-content'>{text}</p>
         </div>
        </div>
    </Card>
);

export default ImageCard;