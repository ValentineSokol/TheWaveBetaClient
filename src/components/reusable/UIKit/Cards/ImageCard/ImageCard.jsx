import React from 'react';
import PropTypes from 'prop-types';
import Card from '../Card/Card';
import TypedHeading from '../../Headings/TypedHeading/TypedHeading';
import Heading from '../../Headings/Heading/Heading';
import './ImageCard.scss';

function ImageCard({
  image, imageAlt, headingStrings, headingSize, text, width,
}) {
  return (
    <Card width={width}>
      <div className="ContentWrapper">
        <img data-testid="image" src={image} alt={imageAlt} />
        <div>
          {
                 Array.isArray(headingStrings)
                   ? <TypedHeading headingStrings={headingStrings} size={headingSize} />
                   : <Heading size={headingSize}>{headingStrings}</Heading>
             }
          <p data-testid="text-content">{text}</p>
        </div>
      </div>
    </Card>
  );
}

ImageCard.propTypes = {
  image: PropTypes.string.isRequired,
  imageAlt: PropTypes.string.isRequired,
  headingStrings: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.arrayOf(PropTypes.string),
  ]).isRequired,
  width: PropTypes.string.isRequired,
  text: PropTypes.string.isRequired,
  headingSize: PropTypes.oneOf(['1', '2', '3', '4', '5', '6']),
};

ImageCard.defaultProps = {
  headingSize: '1',
};
export default ImageCard;
