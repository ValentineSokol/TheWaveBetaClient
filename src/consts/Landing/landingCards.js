import keepCalm from "../../img/landing/keepCalm.png";
import happyCommunity from "../../img/landing/happyCommunity.jpg";
import innovation from "../../img/landing/innovation.jpg";
import writersLove from "../../img/landing/writersLove.jpg";
import ImageCard from "../../components/reusable/UIKit/Cards/ImageCard/ImageCard";
import React from "react";

const landingCards = [
    {
        image: keepCalm,
        imageAlt: 'Keep calm and write fanfiction!',
        text: 'in "The Wave" we don\'t let YOUR creative freedom to be strangled by a clunky UI and laggy back-end! You write fanfiction â€” we handle the rest!',
        headingStrings: 'Focus on what\'s truly important!',
    },
    {
        image: happyCommunity,
        imageAlt: 'Our Community',
        headingStrings: 'Our Community',
        text: 'Our nurturing community will help new authors to really hone their skill and become better! You will make a lot of friends here!'
    },
    {
        image: innovation,
        headingStrings: 'Time for innovation!',
        imageAlt: 'We bring new ideas!',
        text: 'We are eager to bring new ideas to the table! Collaborative writing, Text-To-Speech and much more awaits you here!'
    },
    {
        image: writersLove,
        headingStrings: 'We want you to feel at home with us!',
        imageAlt: 'Love',
        text: 'Built by writers for writers! We have experienced all the painpoints of major fanfiction platforms and this project is a remedy for all of them!'
    },
];
export default  landingCards.map((cardData, index) => <ImageCard width='50%' headingSize='3' key={index} {...cardData} />);