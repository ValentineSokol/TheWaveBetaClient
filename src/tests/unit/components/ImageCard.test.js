    import React from "react";

import ImageCard from "../../../components/reusable/UIKit/Cards/ImageCard/ImageCard";
import Heading from '../../../components/reusable/UIKit/Headings/Heading/Heading';
import landingCards from "../../../consts/Landing/landingCards";
import TypedHeading from "../../../components/reusable/UIKit/Headings/TypedHeading/TypedHeading";



describe("correctly creates a landing card.", () => {
  const [card] = landingCards;
  const component = <ImageCard {...card} />;

  it("renders a component.", () => {
    const renderedComponent = shallow(component);
    expect(renderedComponent).toBeTruthy();
  });
  it("rendered markup matches a snapshot.", () => {
    const renderedMarkup = render(component);
    expect(renderedMarkup).toMatchSnapshot();
  });
  it('Rounds the image based on a \'round\' prop.', () => {
      const findImage = () => renderedComponent.find('img');
      const roundClass = 'RoundImage';
      const renderedComponent = shallow(component);
      expect(findImage().hasClass(roundClass)).toBe(false);
      renderedComponent.setProps({ round: true });
      expect(findImage().hasClass(roundClass)).toBe(true);

  });
  it('Renders Heading, if \'headingStrings\' props is passed a single string.', () => {
      const renderedComponent = shallow(component);
      const foundHeadings = renderedComponent.find(Heading);
      expect(foundHeadings.length).toBe(1);
  });
  it('Renders Heading, if \'headingStrings\' props is passed a single string.', () => {
    const renderedComponent = shallow(component);
    renderedComponent.setProps({ headingStrings: ['a'] });
    const foundHeadings = renderedComponent.find(TypedHeading);
    expect(foundHeadings.length).toBe(1);
  });

});

