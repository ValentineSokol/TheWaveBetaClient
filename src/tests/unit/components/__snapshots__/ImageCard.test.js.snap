// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`correctly creates a landing card. rendered markup matches a snapshot. 1`] = `
initialize {
  "0": Node {
    "attribs": Object {
      "class": "Card",
    },
    "children": Array [
      Node {
        "attribs": Object {
          "class": "ContentWrapper",
        },
        "children": Array [
          Node {
            "attribs": Object {
              "data-testid": "image",
            },
            "children": Array [],
            "name": "img",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": Node {
              "attribs": Object {},
              "children": Array [
                Node {
                  "attribs": Object {
                    "class": "Heading",
                  },
                  "children": Array [],
                  "name": "h1",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": Node {
                    "attribs": Object {
                      "data-testid": "text-content",
                    },
                    "children": Array [],
                    "name": "p",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": null,
                    "parent": [Circular],
                    "prev": [Circular],
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "data-testid": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "data-testid": undefined,
                    },
                  },
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                  },
                },
                Node {
                  "attribs": Object {
                    "data-testid": "text-content",
                  },
                  "children": Array [],
                  "name": "p",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": Node {
                    "attribs": Object {
                      "class": "Heading",
                    },
                    "children": Array [],
                    "name": "h1",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": [Circular],
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "class": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "class": undefined,
                    },
                  },
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "data-testid": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "data-testid": undefined,
                  },
                },
              ],
              "name": "div",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": [Circular],
              "type": "tag",
              "x-attribsNamespace": Object {},
              "x-attribsPrefix": Object {},
            },
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "data-testid": undefined,
            },
            "x-attribsPrefix": Object {
              "data-testid": undefined,
            },
          },
          Node {
            "attribs": Object {},
            "children": Array [
              Node {
                "attribs": Object {
                  "class": "Heading",
                },
                "children": Array [],
                "name": "h1",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": Node {
                  "attribs": Object {
                    "data-testid": "text-content",
                  },
                  "children": Array [],
                  "name": "p",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": [Circular],
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "data-testid": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "data-testid": undefined,
                  },
                },
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
              Node {
                "attribs": Object {
                  "data-testid": "text-content",
                },
                "children": Array [],
                "name": "p",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": Node {
                  "attribs": Object {
                    "class": "Heading",
                  },
                  "children": Array [],
                  "name": "h1",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": [Circular],
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                  },
                },
                "type": "tag",
                "x-attribsNamespace": Object {
                  "data-testid": undefined,
                },
                "x-attribsPrefix": Object {
                  "data-testid": undefined,
                },
              },
            ],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": Node {
              "attribs": Object {
                "data-testid": "image",
              },
              "children": Array [],
              "name": "img",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": [Circular],
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "data-testid": undefined,
              },
              "x-attribsPrefix": Object {
                "data-testid": undefined,
              },
            },
            "type": "tag",
            "x-attribsNamespace": Object {},
            "x-attribsPrefix": Object {},
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": Node {
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
    },
    "prev": null,
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;
